include/alpaka/math/FloatEqualExact.hpp

```diff
@@ -0,0 +1,54 @@
+/* Copyright 2019 Axel Huebl, Benjamin Worpitz
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/Common.hpp>
+
+#include <type_traits>
+
+namespace alpaka
+{
+    namespace math
+    {
+        /** Compare two floating point numbers for exact equivalence. Use only when necessary, and be aware of the
+         * implications. Most codes should not use this function and instead implement a correct epsilon-based
+         * comparison. If you are unfamiliar with the topic, check out
+         * https://www.geeksforgeeks.org/problem-in-comparing-floating-point-numbers-and-how-to-compare-them-correctly/
+         * or Goldberg 1991: "What every computer scientist should know about floating-point arithmetic",
+         * https://dl.acm.org/doi/10.1145/103162.103163
+         *
+         * This function calls the == operator for floating point types, but disables the warning issued by the
+         * compiler when compiling with the float equality warning checks enabled. This warning is valid an valuable in
+         * most codes and should be generally enabled, but there are specific instances where a piece of code might
+         * need to do an exact comparison (e.g. @a CudaVectorArrayWrapperTest.cpp). The verbose name for the function
+         * is intentional as it should raise a red flag if used while not absolutely needed. Users are advised to add a
+         * justification whenever they use this function.
+         *
+         * @tparam T both operands have to be the same type and conform to std::is_floating_point
+         * @param a first operand
+         * @param b second operand
+         * @return a == b
+         */
+        template<typename T>
+        ALPAKA_FN_INLINE ALPAKA_FN_HOST_ACC typename std::enable_if_t<std::is_floating_point<T>::value, bool>
+        floatEqualExactNoWarning(T a, T b)
+        {
```

Comment:
Why the `enable_if` instead of a `static_assert` in the body? I cannot imagine where you would need this function to be used in a SFINAE context where you would want to deselect a template because you cannot call e.g. `floatEqualExactNoWarning` on an `int`.

Reply:
You're right, thanks. No particular reason, probably just the force of habit.